subsys/bluetooth/services/fast_pair/fp_keys.c

```diff
@@ -131,6 +134,7 @@ int fp_keys_decrypt(const struct bt_conn *conn, uint8_t *out, const uint8_t *in)
 		break;
 
 	case FP_STATE_INITIAL:
+	case FP_STATE_WAIT_FOR_PERSONALIZED_NAME:
```

Comment:
We enter this state also in case user simply changes device name on Android phone. In that case, we should be able to encrypt and decrypt messages normally, right? Then maybe we could simply allow both encryption and decryption (at least for now).

Still I agree, that we should consult our doubts (let's do it under separate ticket to speed up development).

Reply:
No, we shouldn't. It's a special flow when key based pairing write clearly indicates that it will be followed by additional data write with data id 0x01 that means personalized name. No other message should be decrypted.

Reply:
Discussed offline. For now we decided to fail to decrypt other messages while we wait for personalized name write in both write flows. We will discuss this approach with Google as a follow-up task.